src/ResourceManager/IotHub/Commands.IotHub/IotHub/RemoveAzureRmIotHubCertificate.cs

```diff
@@ -0,0 +1,73 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System;
+    using System.Management.Automation;
+    using Microsoft.Azure.Management.IotHub;
+
+    [Cmdlet(VerbsCommon.Remove, "AzureRmIotHubCertificate", SupportsShouldProcess = true)]
+    public class RemoveAzureRmIotHubCertificate : IotHubBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Resource Group")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Iot Hub")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = "Name of the Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string CertificateName { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            HelpMessage = "Etag of the Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string Etag { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Name, Properties.Resources.RemoveIotHubCertificate))
+            {
+                try
+                {
+                    this.IotHubClient.Certificates.Delete(this.ResourceGroupName, this.Name, this.CertificateName, this.Etag);
+                }
+                catch (Exception e)
+                {
+                    if (!e.Message.Contains("NotFound"))
```

Comment:
Why the if?

Reply:
@andrew-buckley If we try to delete an object that is already deleted, it'd return a 404/NotFound. We shouldn't return a not found on a delete because the net effect (it's gone) is the same.

Reply:
removed condition